### DISCUSSION TITLE
Initial conditions with TimedSubdomainModifier

### DISCUSSION THREAD

**Discussion by j-wijnen** [idaholab/moose#28616] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi, I am new to MOOSE, but run into some unexpected results.

My problem is related to a welding simulation, demonstrated with a simpler problem here. I have divided my domain into two subdomains (left and right with IDs 0 and 2 ). The right subdomain is not active at the start of the simulation. I am increasing the temperature on the boundary between the two subdomains to 1000. Next, I want to activate the right block with an initial value of 1000 and deactivate the boundary condition, so that the heat can flow from the right domain to the left domain, i.e. the temperature field should even out. I activate the right region by using a TimedSubdomainModifier to move the domain to the block with ID 1, which has an initial condition of 1000, and deactivate the boundary condition (see input file at bottom post).

However, in this case, the initial condition does not seem to be applied. This is the last increment before the activation of the right region (at t=10) :
![image](https://github.com/user-attachments/assets/59f73330-9804-4867-9cb2-5cb8fbec93ae)
This is the increment after the activation:
![image](https://github.com/user-attachments/assets/4ff0f8dc-9fbc-4f1b-bee2-afebdba6ac6c)
Furthermore, it seems like some boundary condition is now applied to the right side, as this is at a later time increment:
![image](https://github.com/user-attachments/assets/c84afbed-f995-4ba2-baa0-2b223260b7b2)

The behavior is different, but still not as expected, if I set the execute_on parameter of the TimedSubdomainModifier to TIMESTEP_END. In this case the initial condition is applied, but the temperature in that region does not evolve. This is the result at a later time increment:
![image](https://github.com/user-attachments/assets/5f51c28b-7342-4b1d-9044-e07698beedcb)
Additionally the small cold region, which is just in the left domain, is introduced.

Any ideas what I am doing wrong? Or is going wrong in MOOSE?

Here is my input file:
```
[Mesh]
  [generated]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 40
    ny = 10
    xmax = 40
    ymax = 10
  []
  [subdomains]
    type = ParsedSubdomainMeshGenerator
    input = generated
    combinatorial_geometry = 'x > 20.0'
    block_id = 2
  []
  [middle_sideset]
    type = SideSetsBetweenSubdomainsGenerator
    input = subdomains
    new_boundary = 'middle'
    primary_block = 0
    paired_block = 2
  []

  add_subdomain_ids = 1
[]

[Variables]
  [temp]
    order = first
    family = LAGRANGE
    block = '0 1'
  []
[]

[ICs]
    [base_ic]
        type = ConstantIC
        variable = temp
        value = 20
        block = 0
    []
    [weld_ic]
        type = ConstantIC
        variable = temp
        value = 1000
        block = 1
    []
[]

[Kernels]
  [heat_conduction]
      type = HeatConduction
      variable = temp
      block = '0 1'
  []
  [heat_timederivative]
      type = HeatConductionTimeDerivative
      variable = temp
      block = '0 1'
  []
[]

[Materials]
    [thermal]
        type = HeatConductionMaterial
        thermal_conductivity = 0.048
        specific_heat = 0.45
    []
    [density]
        type = GenericConstantMaterial
        prop_names = 'density'
        prop_values = 0.0078
    []
[]

[BCs]
  active = 'temp_middle'
  [temp_middle]
    type = FunctionDirichletBC
    variable = temp
    boundary = 'middle'
    function = '20 + 980 * t/10'
  []
[]

[Controls]
    active = 'disableheating'
    [disableheating]
      type = ConditionalFunctionEnableControl
      conditional_function = 't >= 10'
      disable_objects = 'BCs::temp_middle'
      enable_objects = ''
      execute_on = 'INITIAL TIMESTEP_BEGIN'
    []
[]

[UserObjects]
    active = 'insert'
   [insert]
       type = 'TimedSubdomainModifier'
       times =       '10'
       blocks_from = '2'
       blocks_to =   '1'
       execute_on = 'INITIAL TIMESTEP_BEGIN'
       apply_initial_conditions = true
   []
[]

[Problem]
  type = FEProblem
  kernel_coverage_check = false
[]

[Executioner]
  type = Transient
  end_time = 50
  dt = 1
  solve_type = "NEWTON"
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
  l_tol = 1.0e-6
  nl_rel_tol = 1.0e-4
  nl_abs_tol = 1.0e-4
[]

[Outputs]
  exodus = true
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

We actually just changed (wednesday night, released mid Thursday) the behavior of the subdomain modifiers with regards to initial conditions. You probably should update so we are looking at the same simulations.

The initial condition should be applied properly now.
The default bounday condition is 0 flux. Since the BC was applied on the middle and is then turned off, I think that's what you should see

**Reply by j-wijnen** [reply]
Thank you! Indeed it is behaving as expected now.
